src/library/lastplayedcache.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <QObject>
+#include <QSet>
+#include <QSqlDatabase>
+#include <QString>
+
+#include "track/track.h"
+#include "track/trackid.h"
+
+class TrackCollection;
+
+const QString LASTPLAYEDTABLE_NAME = "last_played";
+
+/// LastPlayedCache is a helper object for creating and maintaining
+/// the "last_played" temporary view on the database. It creates the
+/// view and also watches for changes to the Setlog Playlists and updates
+/// tracks in the cache as necessary.
+class LastPlayedCache : public QObject {
+    Q_OBJECT
+  public:
+    LastPlayedCache(TrackCollection* trackCollection);
+    ~LastPlayedCache() {
+    }
+
+    static QDateTime fetchLastPlayedTime(const QSqlDatabase& db, TrackPointer pTrack);
```

Comment:
m_pTrackCollection is a member and already holds a database connection. Why is this a static method that is passed a database connection? Do we need this flexibility?

Reply:
see above -- because trackdao can't get to this object in any way I can see

Reply:
fixed, I split up the object